google/cloud/firestore_v1/async_transaction.py

```diff
@@ -182,18 +188,30 @@ async def get(
                 should be retried.  Defaults to a system-specified policy.
             timeout (float): The timeout for this request.  Defaults to a
                 system-specified value.
+            explain_options
+                (Optional[:class:`~google.cloud.firestore_v1.query_profile.ExplainOptions`]):
+                Options to enable query profiling for this query. When set,
+                explain_metrics will be available on the returned generator.
 
         Yields:
-            .DocumentSnapshot: The next document snapshot that fulfills the
-            query, or :data:`None` if the document does not exist.
+            DocumentSnapshot: The next document snapshot that fulfills the query,
+            or :data:`None` if the document does not exist.
         """
         kwargs = _helpers.make_retry_timeout_kwargs(retry, timeout)
         if isinstance(ref_or_query, AsyncDocumentReference):
+            if explain_options is not None:
+                warnings.warn(
+                    "explain_options not supported in transanction with "
+                    "document references and will be ignored. To use "
+                    "explain_options, use transaction with query instead."
+                )
             return await self._client.get_all(
                 [ref_or_query], transaction=self, **kwargs
             )
         elif isinstance(ref_or_query, AsyncQuery):
-            return await ref_or_query.stream(transaction=self, **kwargs)
+            if explain_options is not None:
+                kwargs["explain_options"] = explain_options
+            return ref_or_query.stream(transaction=self, **kwargs)
```

Comment:
is the drop of `await` intentional here?

Reply:
I think it was unintentional. I'm quite unsure whether we need `await` here: `stream()` itself is not async, but it does call an async function inside, without using `await`. Does this make `stream()` an implied async function?

Reply:
Yeah lookin at the code, it looks like this is calling AsyncQuery.stream, which is an async generator, and should not need to be awaited. But now I'm wondering how the code worked before... Was this a bug? Or am I missing something?

Reply:
Anyway, it seemed ok to undo the change, so I undid it for now.

Reply:
From our discussion: 

You were right that `stream()` should not be awaited, and this was a bug the unit tests did not catch because `stream()` was overwritten by `Mock`. I made the following changes to correct it:
- remove await
- only mock the API for unit tests for `transaction.get(query)` so we can test the whole `stream()` behavior, both sync and async (unit tests of `transaction.get_all()` also need some work but I will keep it out of scope for this PR)
- add async system test to calling `transaction.get(query)` inside the transaction decorator, in addition to `query.get(transaction)`

Reply:
thanks for adding all these extra tests, even though this was an existing bug. Very helpful!